File: blindseoul-backend/src/main/java/com/github/jinhojinho/blindseoul_backend/dto/Blindwalkdto.java
package com.github.jinhojinho.blindseoul_backend.dto;

import lombok.Data;

@Data
public class Blindwalkdto {
    private Long sidewalkId;
    private Integer subId;
    private Double latMin;
    private Double lonMin;
    private Double latMax;
    private Double lonMax;
    private String rnNm;
    private String brllBlkKndCode;
    private String cnstrctYy;
}

File: blindseoul/lib/model/blindwalk_dto.dart
class BlindwalkDto {
  final int sidewalkId;
  final int subId;
  final double latMin;
  final double lonMin;
  final double latMax;
  final double lonMax;
  final String rnNm;
  final String brllBlkKndCode;
  final String cnstrctYy;

  BlindwalkDto({
    required this.sidewalkId,
    required this.subId,
    required this.latMin,
    required this.lonMin,
    required this.latMax,
    required this.lonMax,
    required this.rnNm,
    required this.brllBlkKndCode,
    required this.cnstrctYy,
  });

  factory BlindwalkDto.fromJson(Map<String, dynamic> json) {
    return BlindwalkDto(
      sidewalkId: json['sidewalkId'],
      subId: json['subId'],
      latMin: json['latMin'].toDouble(),
      lonMin: json['lonMin'].toDouble(),
      latMax: json['latMax'].toDouble(),
      lonMax: json['lonMax'].toDouble(),
      rnNm: json['rnNm'],
      brllBlkKndCode: json['brllBlkKndCode'] ?? '',
      cnstrctYy: json['cnstrctYy'] ?? '',
    );
  }
}
